src/HealthGPS/analysis_module.cpp

```diff
@@ -66,7 +106,27 @@ void AnalysisModule::initialise_population(RuntimeContext &context) {
     publish_result_message(context);
 }
 
-void AnalysisModule::update_population(RuntimeContext &context) { publish_result_message(context); }
+void AnalysisModule::update_population(RuntimeContext &context) {
+
+    // The inner map will have keys which are every factor except the two we are analysing
+    // The value of each key will be reset to 0.0
+    std::unordered_map<std::string, double> inner_map{};
+    for (const auto &entry : context.mapping().entries()) {
+        if (entry.key() != context.mapping().entries().at(0).key() &&
+            entry.key() != context.mapping().entries().at(1).key()) {
+            inner_map[entry.key().to_string()] = 0.0;
+        }
+    }
+
+    // Iterate over every element in calculated_factors_ and set its value to inner map
+    for (auto &outer : calculated_factors_) {
+        for (auto &inner : outer.second) {
+            inner.second = inner_map;
+        }
+    }
```

Comment:
Given that this is just a default value, if using `std::vector`(s) you can use `std::fill` to reset the conents.